lite/kernels/opencl/pad2d_image_compute.cc

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include "lite/backends/opencl/cl_half.h"
+#include "lite/backends/opencl/cl_image_converter.h"
+#include "lite/backends/opencl/cl_include.h"
+#include "lite/core/kernel.h"
+#include "lite/core/op_registry.h"
+#include "lite/kernels/opencl/image_helper.h"
+#include "lite/operators/op_params.h"
+#include "lite/utils/logging.h"
+#include "lite/utils/replace_stl/stream.h"
+
+namespace paddle {
+namespace lite {
+namespace kernels {
+namespace opencl {
+class Pad2dCompute : public KernelLite<TARGET(kOpenCL),
+                                       PRECISION(kFP16),
+                                       DATALAYOUT(kImageDefault)> {
+ public:
+  using param_t = operators::Pad2dParam;
+
+  std::string doc() const override {
+    return "Pad2d using cl::Image2D(ImageDefault/RGBA), kFP16";
+  }
+
+  void PrepareForRun() override {
+    pad2d_param_ = param_.get_mutable<param_t>();
+
+    if (pad2d_param_->mode == "constant") {
+      kernel_func_name_ = "pad2d_constant";
+    } else if (pad2d_param_->mode == "reflect") {
+      kernel_func_name_ = "pad2d_reflect";
+    } else if (pad2d_param_->mode == "edge") {
+      kernel_func_name_ = "pad2d_edge";
+    } else {
+      LOG(FATAL) << "Unknown mode type";
```

Comment:
下个PR，可以打印下`pad2d_param_->mode`：
```suggestion
      LOG(FATAL) << "Unknown mode type:" << pad2d_param_->mode;
```

Reply:
ok